version/version.go

```diff
@@ -17,14 +17,106 @@
 // Package version represents the version the project Dragonfly.
 package version
 
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/dragonflyoss/Dragonfly/apis/types"
+)
+
 var (
-	// Version is the version of the project Dragonfly
+	// version is the version of project Dragonfly
+	// populate via ldflags
+	version string
+
+	// revision is the current git commit revision
 	// populate via ldflags
-	Version string
+	revision string
+
+	// buildDate is the build date of project Dragonfly
+	// populate via ldflags
+	buildDate string
+
+	// goVersion is the running program's golang version
+	goVersion = runtime.Version()
+
+	// os is the running program's operating system
+	os = runtime.GOOS
+
+	// arch is the running program's architecture target
+	arch = runtime.GOARCH
 
 	// DFDaemonVersion is the version of dfdaemon
-	DFDaemonVersion = Version
+	DFDaemonVersion = version
 
 	// DFGetVersion is the version of dfget
-	DFGetVersion = Version
+	DFGetVersion = version
+
+	// SupernodeVersion is the version of supernode
+	SupernodeVersion = version
+
+	// DFVersion the global instance of DragonflyVersion
+	DFVersion *types.SupernodeVersion
```

Comment:
I think  you should make a consideration about the relationship between SupernodeVersion and DragonflyVersion.

Reply:
I still think DragonflyVersion is better because dfdaemon and dfget will this struct too.

Reply:
But DfDaemon and Supernode may be different version. Because there are in different binaries.
For that, separated SuperNodeVersion and DfDaemonVersion may be more reasonable. @yeya24

Reply:
Only when building with `make build`, is it a normal building process because we pass some ldflags in. The three components will be built at the same time and they are in the same version.  Did you mean in the procedure of running  `make build`, the dfdaemon and supernode may have different version? If that, I dont think it is reasonable.